protocol-units/bridge/service/src/chains/ethereum/grpc_handler.rs

```diff
@@ -0,0 +1,158 @@
+use super::{client::EthClient, types::EthAddress};
+use crate::{chains::bridge_contracts::BridgeContract, types::BridgeTransferId};
+use bridge_grpc::{
+	bridge_server::Bridge, BridgeTransferDetailsResponse, GetBridgeTransferDetailsRequest,
+};
+use tonic::{Request, Response, Status};
+
+// Define the service struct that wraps EthClient
+pub struct GRPCServer {
+	eth_client: EthClient,
+}
+
+impl GRPCServer {
+	pub fn new(eth_client: EthClient) -> Self {
+		Self { eth_client }
+	}
+
+	#[cfg(test)]
+	async fn new_for_test() -> Self {
+		use super::client::Config;
+
+		Self {
+			eth_client: EthClient::new(Config::build_for_test())
+				.await
+				.expect("Failed to create EthClient"),
+		}
+	}
+}
+
+// More Admin only utility functions can be added to get more information.
+// For example list of ongoing tasks, list of completed tasks, etc.
```

Comment:
Will the gRPC server be for read-only activity?

Probably good for @musitdev to review the gRPC code, as I'm a bit unfamiliar with its role in the CLI.